### PR TITLE
deprecate EC_POINTs_mul function

### PR DESCRIPTION
**Pull request by bbbrumley**
Reiterating from #8332:

* Deprecate `EC_POINTs_mul`. The `aG + bP` interface of `EC_POINT_mul` is enough for crypto.

Since I started poking at the `ec` module in 2007, `test/ectest.c` is the only linking application I've ever seen using `EC_POINTs_mul`.

I started out trying to remove the calls from `ectest.c` but the tests are honestly such a mess, and those parts that exercise `EC_POINTs_mul` have not been updated in forever. So I chickened out and used the suggestion from @mattcaswell. I'll revisit those tests (hopefully with more deprecations) after 3.0. Especially when I have time to change the internal `mul` interface in `EC_METHOD`.

### CODE REVIEW

**Comment by bbbrumley** [comment]
Oh I see from the CI failure -- lots of checks in `EC_POINTs_mul`. Let me take care of that now.

**Comment by mattcaswell** [comment]
Looks like Travis still has some relevant failures.

**Comment by bbbrumley** [comment]
> Looks like Travis still has some relevant failures.

I mean it's clear it will fail if you try to compile without the deprecated symbols. I guess I was hoping `# OPENSSL_SUPPRESS_DEPRECATED` and the test framework would somehow figure it out.

I don't really know what to do.

Should I wrap the deprecated calls somehow with `ifdef` blocks? (Maybe the tests will still function, I'm not sure. It's a bunch of spaghetti testing right now so it is hard to extract chunks.)

**Comment by mattcaswell** [comment]
> I mean it's clear it will fail if you try to compile without the deprecated symbols. I guess I was hoping `# OPENSSL_SUPPRESS_DEPRECATED` and the test framework would somehow figure it out

`OPENSSL_SUPPRESS_DEPRECATED` just suppresses compiler warnings if you attempt to use deprecated functions. 

You need to separately decide what to do if the symbols are not present at all. Options are:

1) Don't compile or run the entire test in that case (seems a bit over the top)
2) Rewrite the code using the deprecated interface to not use it
3) Don't compile the bits of the test that use the deprecated API

Of these (2) is the preferred option. The main reason for keeping the old function call there would be if the entire purpose of the test is to test the deprecated function (in which case do (3)). If the use of the deprecated function is incidental to the thing being tested, then we should rewrite it.

**Comment by bbbrumley** [comment]
> You need to separately decide what to do if the symbols are not present at all

Got it -- thanks @mattcaswell . Unfortunately in this case the test functions are mixed between "normal" testing and also directly testing this deprecated functionality. So I don't think (2) is an option given the time frame. Maybe I can make (3) work. I'll have a go this evening.

**Comment by bbbrumley** [comment]
> Looks like Travis still has some relevant failures.

OK that wasn't as hackish as I though it would be. Fixed in c3c7060c589852494adffced6cbb1d467b98e523 hopefully.

**Comment by bbbrumley** [comment]
Haha so mixed declaration and code failures -- I obviously don't even understand what that _really_ means because:

https://github.com/openssl/openssl/blob/master/test/ectest.c#L66-L68

and that's why I moved those in the first place. Anyway I'll take a look now ...

File: crypto/err/openssl.txt [hunk 1]
```diff
@@ -696,6 +696,7 @@ EC_F_EC_POINT_INVERT:210:EC_POINT_invert
 EC_F_EC_POINT_IS_AT_INFINITY:118:EC_POINT_is_at_infinity
 EC_F_EC_POINT_IS_ON_CURVE:119:EC_POINT_is_on_curve
 EC_F_EC_POINT_MAKE_AFFINE:120:EC_POINT_make_affine
+EC_F_EC_POINT_MUL:309:
```
Comment (bbbrumley) [inline crypto/err/openssl.txt:699]: @mattcaswell out of curiosity, what is going on here after `make update`? I'm not great with the error side of OpenSSL, but looking at the diff I'd expect the function name on the RHS.

Comment (richsalz) [reply crypto/err/openssl.txt:699]: OpenSSL no longer displays the function name, so that field is now blank.

**Comment by mattcaswell** [comment]
Pushed. Thanks!

**Pull request closed without merging**